### PR TITLE
Working edit commands

### PR DESCRIPTION
**Pull request by archthegit**
- fixed insert command bug

### CODE REVIEW

File: src/struct_command_manager.ts [hunk 1]
```diff
@@ -152,12 +166,59 @@ export class StructCommandManager {
...
+        /* for achu to do */
+        vscode.window.showInformationMessage("COMMAND SAID: "+command);
+        var arr = command.split(" ");
+        var num_to_delete = parseInt(arr[1]);
+
+        var temp = this.stay_speech.split(" ");
+        for (var i=1;i<=num_to_delete;i++){
+            temp.splice(temp.length-i,1);
+        }
+        var commented_speech = "#string \"" + temp.join(" ")+ " *stay*\";;"
```
Comment (lawwman) [inline src/struct_command_manager.ts]: *stay* should not be there anymore

File: src/struct_command_manager.ts [hunk 2]
```diff
@@ -36,6 +36,12 @@ export class StructCommandManager {
     edit_stack: edit_stack_item[];
 
     debugMode: boolean;
+    /* is true when command is being held */
+    holding: boolean;
+
+    stay_speech: string;
```
Comment (lawwman) [inline src/struct_command_manager.ts]: not sure what stay speech is for. just need to backspace only

File: src/struct_command_manager.ts [hunk 3]
```diff
@@ -339,13 +395,26 @@ export class StructCommandManager {
...
+        this.curr_speech = [""];
+
         var copiedStructCommand = this.deepCopyStructCommand()
         var copiedSpeechHist = this.deepCopySpeechHist()
         var oldIdx = this.curr_index;
 
-        if (this.curr_index < amt_to_remove) this.curr_index = start_pos;
-        else this.curr_index -= amt_to_remove;
-
+        var lostCursor = false;
```
Comment (lawwman) [inline src/struct_command_manager.ts:404]: haha what isn't this code ive written. but why is it just added now

File: src/struct_command_manager.ts [hunk 4]
```diff
@@ -152,12 +164,56 @@ export class StructCommandManager {
...
+        vscode.window.showInformationMessage("COMMAND SAID: "+command);
+        var arr = command.split(" ");
+        var num_to_delete = parseInt(arr[1]);
+        
+        var temp = this.curr_speech.join(" ").split(" ");
+        for (var i=1;i<=num_to_delete;i++){
+            temp.splice(temp.length-i,1);
+        }
+        var commented_speech = "#string \"" + temp.join(" ")+"\";;"
+        this.struct_command_list.splice(this.curr_index,1,commented_speech);
```
Comment (lawwman) [inline src/struct_command_manager.ts:187]: still looks abit weird

File: src/struct_command_manager.ts (hunk 3 above)
Comment (archthegit) [reply src/struct_command_manager.ts:404]: I dont know lol I didnt do anything to this function

File: src/struct_command_manager.ts (hunk 4 above)
Comment (archthegit) [reply src/struct_command_manager.ts:187]: what do you mean?

**Pull request merged**